src/rpc/mining.cpp

```diff
@@ -149,9 +153,52 @@ UniValue getnewblock(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_INVALID_PARAMS, "required_wait must be non-negative.");
     }
 
+    const std::string xfieldParam = request.params[2].isNull() ? "" : request.params[2].get_str();
+    TAPYRUS_XFIELDTYPES xfieldType = TAPYRUS_XFIELDTYPES::NONE;
+    xFieldInput xfield;
+    xfield.invalid = true;
+
+    if (!request.params[2].isNull())
+    {
+        //using lambda to avoid temp variables
+        xfieldType = [xfieldParam](int splitAt, TAPYRUS_XFIELDTYPES max) -> TAPYRUS_XFIELDTYPES
+            { int x = splitAt > 0 ? atoi(xfieldParam.substr(0,splitAt)) : 0;
+            return x > 0 && x < int(max) ? TAPYRUS_XFIELDTYPES(x) : TAPYRUS_XFIELDTYPES::NONE;
+            } (xfieldParam.find(':'), TAPYRUS_XFIELDTYPES::MAX_XFIELDTYPE );
+
+        if(xfieldType ==  TAPYRUS_XFIELDTYPES::NONE)
+        {
+            throw JSONRPCError(RPC_INVALID_PARAMS, "xfield_type parameter could not be parsed. Check if the xfield parameter has format: <xfield_type:new_xfield_value>.");
+        }
+
+        switch(xfieldType){
+            case TAPYRUS_XFIELDTYPES::AGGPUBKEY:
+            {
+                std::string aggPubkeyString = xfieldParam.substr(xfieldParam.find(':')+1);
+                if (IsHex(aggPubkeyString)) {
+                    std::vector<unsigned char> data = ParseHex(aggPubkeyString);
+                    CPubKey aggPubKey(data);
+                    if (aggPubKey.IsFullyValid() && aggPubKey.IsCompressed())
+                    {
+                        xfield.xFieldValue.aggPubKey = std::vector<unsigned char>(data.begin(), data.end());
+                        xfield.invalid = false;
+                        break;
+                    }
+                }
+                xfield.invalid = true;
+            }
+            break;
+            default:
+                xfield.invalid = true;
+        }
+
+        if (xfield.invalid == true) {
+            throw JSONRPCError(RPC_INVALID_PARAMS, "xfield parameter was invalid. It is expected to be <xfield_type:new_xfield_value>.");
```

Comment:
It would be useful to know from the error message why it is invalid. From the above checks, the following possibilities are possible

* Invalid public key
* Uncompressed public key
* Unsupported xfield type